tests/test_constants.py

```diff
@@ -40,3 +42,9 @@ def test_use_emjois(monkeypatch, windows, USE_EMOJI, encoding, expected):
         if USE_EMOJI is not None:
             monkeypatch.setenv("USE_EMOJI", USE_EMOJI)
         assert use_emjois() is expected
+
+
+def test_default_python(monkeypatch):
+    monkeypatch.setattr(constants, "DEFAULT_PYTHON", "bad_python")
```

Comment:
Should this be `PIPX_DEFAULT_PYTHON`?

Reply:
I'm not sure, but I don't think so. The way constants are set up, environment variables are read before the tests are run, so patching the environment variable itself (`PIPX_DEFAULT_PYTHON`) wouldn't do anything here. So, we need to patch the python variable that holds the environment variable instead.

And I'm fairly new to your code base, so I might be wrong, but it seems to me this is the same way the other constants are handled in the test suite (like patching `LOCAL_BIN_DIR` instead of `PIPX_BIN_DIR` [here](https://github.com/pipxproject/pipx/blob/master/tests/conftest.py#L21)).

Reply:
I think we could do it with `importlib.reload(pipx.constants)` but I'm not sure if there would be side-effects.